### PR TITLE
chore(deps): September 2024 bump

### PR DESCRIPTION
**Pull request by jailln**
## Description

Update all dependencies (when possible).

Previous major bump in https://github.com/iTowns/itowns/pull/2342

## Peer dependencies

- [x] `three` (`0.165.0` to `0.168.0`)

~~A bug appears in `0.167.1`: Zooming in a view with a perspective camera does not work anymore. I've narrowed the problem down to the `buffer` computed in [View.getPickingPositionFromDepth](https://github.com/iTowns/itowns/blob/c1409c509328c02aa378ec2749a6e3d9b0885ba5/src/Core/View.js#L1030) which is different with `three` `0.166.1` and `0.167.1`. 
I also found the responsible commit in `threejs` using `yarn link` and `git bisect`. It comes from the [following PR](https://github.com/mrdoob/three.js/pull/28789) (note that the commit hash is `a38bbdd6fbe971049f2083d79bbca2cda7972a44` and not the one in the PR). I haven't found out the buug yet, it's quite low level :sweat_smile: cc @Desplandis if you have any ideas~~ => Solved

- [x] `proj4` (`2.11.0` to `2.12.1`): No breaking changes

## Dependencies

- [ ]  `@mapbox/mapbox-gl-style-spec`  (`13.28.0`): Still the same problems as #2342 . Won't update here, see #2247 and #2369.
- [x] `@tweenjs/tween.js` (`23.1.2` to `25.0.0`): Updates to `StreetControls` and `CameraUtils` to use the new way to add tweens to the tween Group.
- [x] `shpjs` (`6.0.1` to `6.1.0`): No breaking changes

## Developer dependencies
- [x] `@babel/cli`, `@babel/core` and the following plug-ins: `@babel/plugin-transform-runtime` and `@babel/preset-env`.
- [x] `babel-loader`
- [x] `chart.js`
- [ ] conventional-changelog-cli: same as #2342 - Pure ESM package, issues with our CJS script.
- [x] `core-js`
 - [ ] `eslint`: same as #2342 - cannot update to ESLint 9 because `eslint-plugin-import` requires ESLint8.
- [x] `eslint-plugin-import`
- [x] `eslint-webpack-plugin`
- [x] `mocha`
- [x] `node-fetch`: removed since node 18 natively supports the fetch API.
- [x] `puppeteer`
- [x] `replace-in-file`
- [x] `sinon`
- [x] `typescript`
- [x] `webpack`
- [x] `webpack-dev-server`
- [x] `@xmldom/xmldom`

~~I've also noticed that the [Fetcher tests](https://github.com/iTowns/itowns/blob/c1409c509328c02aa378ec2749a6e3d9b0885ba5/test/unit/fetcher.js) are not run by the CI and fail locally (same on master). I haven't look at why they don't run on the CI yet, if someone knows that would be greatly appreciated :) The local failure is quite weird: it only fails un run mode but succeeds in debug mode :sweat_smile:~~ => Solved

### CODE REVIEW

**Comment by Desplandis** [comment]
I'm all (in a near future) for removing scripts that are maybe not necessary anymore. That includes:
- The `_BATCHID` substitution script for `GLTFLoader`
- The copying of `three/examples/jsm` scripts into the `ThreeExtended` directory. We could relax the linting rules on `.js` extension and let the bundle bundle all ESM modules together.

This could remove 3/4 dev dependencies in my opinion.

**Comment by Desplandis** [comment]
For the three.js issue (thanks to your bisect), we can assume that since the packing in GPU from depth to RGBA has changed, our unpack in CPU (see the code below) is no longer valid...

https://github.com/iTowns/itowns/blob/e38663701ff8f7ddefedb0721bb61fa7c28078db/src/Renderer/LayeredMaterial.js#L13-L23

**Comment by Desplandis** [comment]
Weird for the `test/unit/fetcher.js` test, it works locally for me.

**Comment by jailln** [comment]
> I'm all (in a near future) for removing scripts that are maybe not necessary anymore. That includes:
> 
> * The `_BATCHID` substitution script for `GLTFLoader`
> * The copying of `three/examples/jsm` scripts into the `ThreeExtended` directory. We could relax the linting rules on `.js` extension and let the bundle bundle all ESM modules together.
> 
> This could remove 3/4 dev dependencies in my opinion.

Totally agree with you! 

> For the three.js issue (thanks to your bisect), we can assume that since the packing in GPU from depth to RGBA has changed, our unpack in CPU (see the code below) is no longer valid...

Yes, that was it! Thanks to you and @AnthonyGlt for finding that

> Weird for the test/unit/fetcher.js test, it works locally for me.

It was related to the node version. The test works locally for me when I switch to node 18. However it is still not run by the ci but this is not related to this PR.

File: src/Controls/StreetControls.js [hunk 1]
```diff
@@ -310,6 +310,8 @@ class StreetControls extends FirstPersonControls {
             })
             .start();
 
+        this.tweenGroup.add(this.tween);
+
```
Comment (Desplandis) [inline src/Controls/StreetControls.js:314]: From `tween.js` [changelog](https://github.com/tweenjs/tween.js/releases/tag/v24.0.0):

> Tweens are no longer automatically added or removed from groups by default when you call any Tween methods such as start(), stop(), or pause(), and the preserve parameter to Group.update() now defaults to true and is deprecated to be removed in a future major version.

As such, tweens are not removed in the group and updated continuously (they were automatically removed before the update).

**Review by Desplandis — CHANGES_REQUESTED** [review]

File: src/Controls/StreetControls.js (hunk 1 above)
Comment (jailln) [reply src/Controls/StreetControls.js:314]: Bien vu; done :)

**Review by Desplandis — APPROVED** [review]
Thanks for the fixes =)

**Pull request merged**